pipelines/test/ldd-check.yaml

```diff
@@ -0,0 +1,59 @@
+name: ldd-check
+
+needs:
+  packages:
+    - busybox
+    - posix-libc-utils
+
+inputs:
+  files:
+    description: |
+      The files to run `ldd` on and check for missing deps.
+    required: true
```

Comment:
Thanks for doing this! Is there a way to use this interface w/o knowing the list of files in advance? I'd coincidentally added the following test to a package in my local tree:

```yaml
test:
  environment:
    contents:
      packages:
        - apk-tools
        - posix-libc-utils # for ldd
    environment:
      LD_LIBRARY_PATH: /usr/local/cuda-12.5/lib64:/usr/local/cudnn-9/lib64:/opt/tensorrt_llm/libs:/opt/tritonserver/lib
  pipeline:
    - name: Scan .so files for undetected runtime dependencies
      runs: |
        ret=0
        apk info -L "${{package.name}}" | grep \.so$ | sed 's,^,/,' | while read obj; do
          echo -n "Scanning $obj... "
          if ldd "$obj" | grep "not found"; then
            echo "failed"
            ret=1
            continue
          fi
          echo "ok"
        done
        exit $ret
```

Here, I don't know the `.so*` files in advance, so I'm just searching the entire package. Ultimately, I think we should teach `melange` to `shlibdeps` these on its own, allowing users to add exceptions.

Reply:
I didn't do that, but I wouldn't be opposed to adding the functionality. 

I agree with the fact that something else other than crappy shell scripts should do this.

Reply:
Also, I explicitly did not allow wildcards, but I'm not opposed to that either.

Reply:
> Thanks for doing this! Is there a way to use this interface w/o knowing the list of files in advance? I'd coincidentally added the following test to a package in my local tree:
> 
> ```yaml
> test:
>   environment:
>     contents:
>       packages:
>         - apk-tools
>         - posix-libc-utils # for ldd
>     environment:
>       LD_LIBRARY_PATH: /usr/local/cuda-12.5/lib64:/usr/local/cudnn-9/lib64:/opt/tensorrt_llm/libs:/opt/tritonserver/lib
>   pipeline:
>     - name: Scan .so files for undetected runtime dependencies
>       runs: |
>         ret=0
>         apk info -L "${{package.name}}" | grep \.so$ | sed 's,^,/,' | while read obj; do
>           echo -n "Scanning $obj... "
>           if ldd "$obj" | grep "not found"; then
>             echo "failed"
>             ret=1
>             continue
>           fi
>           echo "ok"
>         done
>         exit $ret
> ```
> 
> Here, I don't know the `.so*` files in advance, so I'm just searching the entire package. Ultimately, I think we should teach `melange` to `shlibdeps` these on its own, allowing users to add exceptions.

fyi: https://github.com/chainguard-dev/internal-dev/issues/7609